pymbolic/geometric_algebra/__init__.py

```diff
@@ -1113,14 +1133,14 @@ def map(self, f):
 # }}}
 
 
-def componentwise(f, expr):
+def componentwise(f: Callable[[CoeffT], CoeffT], expr: T) -> T:
```

Comment:
Should this be more general somehow? `f: Callable[[CoeffT], CoeffR]` or something? No idea..

Reply:
`CoeffT` is already pretty broad, so I would say let's wait until there's a need.